data/transactions/logic/eval.go

```diff
@@ -425,6 +443,14 @@ func (ep *EvalParams) computeAvailability() *resources {
 	return available
 }
 
+func (cx *EvalContext) minAvmVersionCheck() error {
+	if cx.version < cx.EvalParams.minAvmVersion {
+		return fmt.Errorf("program version must be >= %d for this transaction group, but have version %d",
+			cx.minAvmVersion, cx.version)
+	}
+	return nil
+}
```

Comment:
broken out from `versionCheck`

Reply:
what if you made a method func for evalcontext called setVersion you can call during cx "construction time" that sets the context version and PC given a program and proto if it conforms to the four version requirements, and get rid of versionCHeck and minAvmVersionCheck.

Reply:
So basically put all of this:
```
	version, vlen, err := versionCheck(program, cx.Proto)

	cx.version = vlen
	cx.pc = pc
...
...
	if err := cx.minAvmVersionCheck(); err != nil {
		return false, err
	}
```
into a cx method?

See if you like https://github.com/algorand/go-algorand/pull/5528/commits/ed59b5daef014f8ce72df9e0f5d7b2b5dbe9fda9